test/function_test.py

```diff
@@ -682,3 +682,21 @@ def test_mount_deps_have_ids(client, servicer, monkeypatch, test_dir):
     function_create = ctx.pop_request("FunctionCreate")
     for dep in function_create.function.object_dependencies:
         assert dep.object_id
+
+
+def test_no_state_reuse(client, servicer, supports_dir):
```

Comment:
Added an additional test here that reproduced the subtle state reuse issue we previously only detected only in integration tests